### PR TITLE
Add issue templates

### PR DESCRIPTION
**Pull request by dkhalanskyjb**

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/design-considerations.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+---
+name: Design considerations
```
Comment (qwwdfsad) [inline .github/ISSUE_TEMPLATE/design-considerations.md:2]: "Design" in the sense of API semantics is something we got used to, but this term is unlikely to be widespread and approachable to others.

Maybe "Feature request" or "Feature or behaviour request" will suit better?

Comment (dkhalanskyjb) [reply .github/ISSUE_TEMPLATE/design-considerations.md:2]: There's already "feature request" for new features, see `feature_request.md`. "Design considerations" is for changing the existing things, like "`Dispatchers.Unconfined` shouldn't use an event loop", or something like https://github.com/Kotlin/kotlinx.coroutines/issues/2860. Deeply technical discussions about issues with things that are already there.

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+---
+name: Feature request
+about: We're missing something?
+title: ''
+labels: ''
```
Comment (qwwdfsad) [inline .github/ISSUE_TEMPLATE/feature_request.md:5]: Can be `enhancement`, but no hard opinion on this one

**Review by qwwdfsad — APPROVED** [review]
Great job, amazing attention to details (i.e. examples and non-examples) 🔥

**Pull request merged**